src/test/addrman_tests.cpp

```diff
@@ -0,0 +1,522 @@
+// Copyright (c) 2012-2013 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+#include "addrman.h"
+#include "test/test_pivx.h"
+#include <string>
+#include <boost/test/unit_test.hpp>
+#include <crypto/common.h> // for ReadLE64
+
+#include "hash.h"
+#include "random.h"
+
+using namespace std;
```

Comment:
even though this is pulled from upstream, I would prefer if we didn't rely on importing `std` into the global namespace as per the developer guidelines.